src/main/java/org/jvnet/hudson/plugins/shelveproject/ShelveProjectAction.java

```diff
@@ -27,11 +27,11 @@ public ShelveProjectAction(Item item) {
 
   @Override
   public String getIconFileName() {
-    return getShelveIconPath();
+    return getShelveIconPath(getItem());
   }
 
-  private static String getShelveIconPath() {
-    return Jenkins.getInstance().hasPermission(SHELVE_PERMISSION) ? ACTION_ICON_PATH : null;
+  private String getShelveIconPath(Item item) {
+    return item != null && item.hasPermission(SHELVE_PERMISSION) ? ACTION_ICON_PATH : null;
```

Comment:
I would annotate the item parameter of the constructor with `@NonNull` , then null checks are not needed.